### PR TITLE
Separate Event Loop from DefaultTestSelector class

### PR DESCRIPTION
**Pull request by ConorPKeegan**
# Changes
* Removed the event loop from DefaultTestSelector class
* Added a BaseEventLoop class to represent the common attributes of event loops
* Added an EventLoop class for the separated event loop, with the common event loop attributes inherited from the Base class
* Added a factory to generate an event loop from a name and a configuration

# Implements
* Implements enhancement #112

### CODE REVIEW

**Comment by ConorPKeegan** [comment]
@PrzemekWirkus

File: mbed_host_tests/host_tests_event_loop/base_event_loop.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+from Queue import Empty as QueueEmpty   # Queue here refers to the module, not a class
+
+from time import time
+from abc import ABCMeta, abstractmethod
+
+class BaseEventLoop():
+    __metaclass__ = ABCMeta
+
+    def __init__(self, logger, event_queue, dut_event_queue, process, test_selector):
```
Comment (PrzemekWirkus) [inline mbed_host_tests/host_tests_event_loop/base_event_loop.py]: Please add doxygen describing parameters of this ctor.

**Comment by PrzemekWirkus** [comment]
@ConorPKeegan Please redirect this PR from `master` branch to `devel_framework_refactoring`. We will use `devel_framework_refactoring` branch as development branch for refactoring you are doing!

**Comment by ConorPKeegan** [comment]
Replaced by #114, to change the destination branch

**Pull request closed without merging**